src/Views/MainView.vala

```diff
@@ -180,9 +145,46 @@ namespace SwitchboardPlugUserAccounts.Widgets {
             show_all ();
         }
 
+        private void remove_user () {
+            var permission = get_permission ();
+            if (!permission.allowed) {
+                try {
+                    permission.acquire ();
+                } catch (Error e) {
+                    if (!e.matches (GLib.IOError.quark (), GLib.IOError.CANCELLED)) {
+                        var message_dialog = new Granite.MessageDialog.with_image_from_icon_name (
+                            _("Unable to acquire permission"),
+                            _("An account cannot be removed without the required system permission."),
+                            "dialog-password",
+                            Gtk.ButtonsType.CLOSE
+                        ) {
+                            badge_icon = new ThemedIcon ("dialog-error"),
+                            transient_for = (Gtk.Window) get_toplevel ()
+                        };
+                        message_dialog.show_error_details (e.message);
+                        message_dialog.present ();
+                        message_dialog.destroy ();
```

Comment:
Since `present` is non-blocking, you have to do `response_connect` here or the dialog will just immediately destroy